pkg/publicapi/endpoint/orchestrator/job.go

```diff
@@ -344,3 +347,73 @@ func (e *Endpoint) jobResults(c echo.Context) error {
 		Results: resp.Results,
 	})
 }
+
+// godoc for Orchestrator JobLogs
+//
+// @ID				orchestrator/logs
+// @Summary			Displays the logs for a current job/execution
+// @Description		Shows the output from the job specified by `id`
+// @Description		The output will be continuous until either, the client disconnects or the execution completes.
+// @Tags			Orchestrator
+// @Accept			json
+// @Produce			json
+// @Param			id				path	string	true	"ID to get the job logs for"
+// @Param			execution_id	query 	string	false	"Fetch logs for a specific execution"
+// @Param			with_history	query	bool	false	"Fetch historical logs"
+// @Param			follow			query	bool	false	"Follow the logs"
+// @Success		200			{object}	string
+// @Failure		400			{object}	string
+// @Failure		500			{object}	string
+// @Router			/api/v1/orchestrator/jobs/{id}/logs [get]
+func (e *Endpoint) logs(c echo.Context) error {
+	ws, err := upgrader.Upgrade(c.Response(), c.Request(), nil)
+	if err != nil {
+		return fmt.Errorf("failed to upgrade websocket connection: %w", err)
+	}
+	defer ws.Close()
+
+	err = e.logsWS(c, ws)
+	if err != nil {
+		err = ws.WriteJSON(concurrency.AsyncResult[models.ExecutionLog]{
+			Err: err,
+		})
+		if err != nil {
+			c.Logger().Errorf("failed to write error to websocket: %s", err)
+		}
+	}
+	_ = ws.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
+	return nil
+}
+
+func (e *Endpoint) logsWS(c echo.Context, ws *websocket.Conn) error {
+	jobID := c.Param("id")
+	var args apimodels.GetLogsRequest
+	if err := c.Bind(&args); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
+	}
+	if err := c.Validate(&args); err != nil {
+		return err
+	}
+
+	logstreamCh, err := e.orchestrator.ReadLogs(c.Request().Context(), orchestrator.ReadLogsRequest{
+		JobID:       jobID,
+		ExecutionID: args.ExecutionID,
+		WithHistory: args.WithHistory,
+		Follow:      args.Follow,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to open log stream for job %s: %w", jobID, err)
+	}
+
+	for {
```

Comment:
purely a style nit, but slight preference for:
```suggestion
	for logMsg := range logstreamCh {
		if err = ws.WriteJSON(logMsg); err != nil {
		        return err
		}
	}
```

Reply:
Fixed in https://github.com/bacalhau-project/bacalhau/pull/3354/commits/80d1eef8857a74bd9ceb967d9958041bbcd42308